Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/testresult/pending/SubmissionTestResultPendingFragment.kt

```diff
@@ -119,30 +118,28 @@ class SubmissionTestResultPendingFragment : Fragment(R.layout.fragment_submissio
         }
     }
 
-    private fun handleError(exception: CwaWebException) {
-        errorDialog = when (exception) {
-            is CwaClientError, is CwaServerError -> {
-                DialogHelper.showDialog(buildErrorDialog(exception))
-            }
-            else -> {
-                DialogHelper.showDialog(genericErrorDialog)
-            }
+    private fun handleError(exception: Throwable) {
+        val dialogInstance = when (exception) {
+            is CwaClientError, is CwaServerError -> networkErrorDialog
+            else -> genericErrorDialog
         }
+        errorDialog = DialogHelper.showDialog(dialogInstance)
     }
 
     private fun navigateToMainScreen() {
         popBackStack()
     }
 
-    private fun buildErrorDialog(exception: CwaWebException) = DialogHelper.DialogInstance(
-        requireActivity(),
-        R.string.submission_error_dialog_web_generic_error_title,
-        R.string.submission_error_dialog_web_generic_network_error_body,
-        R.string.submission_error_dialog_web_generic_error_button_positive,
-        null,
-        true,
-        ::navigateToMainScreen
-    )
+    private val networkErrorDialog: DialogHelper.DialogInstance
```

Comment:
Are these two error dialogs identical? There used to be the http code included in the web exception, but it is not there. Maybe the generic one is sufficient anyway.

Reply:
The body is different